tests/test_queries.py

```diff
@@ -495,6 +495,59 @@ def test_select_non_projected(self):
         self.assertEqual(list(ret), [{"bar": 2}])
 
 
+class TestSaveJson(BaseSystemTest):
+
+    """ Tests that involve saving the results into a JSON file """
+
+    def setUp(self):
+        super(TestSaveJson, self).setUp()
+        self.test_file_name = 'out.json'
+        self.engine.output_json_as_list = True
+
+    def test_no_results(self):
+        """ JSON output with no results """
+        self.make_table()
+        self.query("SCAN * FROM foobar SAVE " + self.test_file_name)
+        contents = self._get_file_contents()
+        self.assertEqual(contents, "[]\n")
+
+    def test_one_result(self):
+        """ JSON output with one result """
+        self.make_table()
+        self.query("INSERT INTO foobar (id, bar) VALUES ('a', 1)")
+        self.query("SCAN * FROM foobar SAVE " + self.test_file_name)
+        contents = self._get_file_contents()
+        self.assertEqual(contents, '[{"bar":1,"id":"a"}]\n')
+
+    def test_two_results(self):
+        """ JSON output with two results """
+        self.make_table()
+        self.query("INSERT INTO foobar (id, bar) VALUES ('a', 1), ('b', 2)")
+        self.query("SCAN * FROM foobar SAVE " + self.test_file_name)
+        contents = self._get_file_contents()
+        self.assertEqual(
+            contents,
+            '[{"bar":2,"id":"b"},\n{"bar":1,"id":"a"}]\n'
+        )
+
+    def test_three_results(self):
```

Comment:
I went up to three to ensure there's a case where the opening and closing bracket are separted by at least one line in the middle.